scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/TerminationData.java

```diff
@@ -151,41 +155,43 @@ boolean taskTerminated(JobId j, String taskName) {
     void handleTermination(final SchedulingService service) throws IOException, ClassNotFoundException {
         for (TaskTerminationData taskToTerminate : tasksToTerminate.values()) {
             RunningTaskData taskData = taskToTerminate.taskData;
-            Map<String, String> genericInformation = new HashMap<>();
-            VariablesMap variables = null;
-            if (taskToTerminate.internalJob != null) {
-                genericInformation = taskData.getTask().getRuntimeGenericInformation();
-            }
-            try {
-                variables = getStringSerializableMap(service, taskToTerminate);
-            } catch (Exception e) {
-                logger.error("Exception occurred, fail to get variables into the cleaning script: ", e);
-            }
-            try {
-                if (!taskToTerminate.normalTermination) {
-                    taskData.getLauncher().kill();
+            if (!taskToTerminate.killedWhileNotRunning()) {
```

Comment:
the condition name is confused, maybe it s more clear the reverse logic.. something like if (taskToTerminate.killedWhileRunning())